File: 1768.rs
pub fn merge_alternately(word1: String, word2: String) -> String {
    let mut merged = String::new();
    let mut first = word1.chars().peekable();
    let mut last = word2.chars().peekable();

    while first.peek().is_some() || last.peek().is_some() {
        if let Some(first) = first.next() {
            merged.push(first);
        }
        if let Some(last) = last.next() {
            merged.push(last);
        }
    }

    merged
}

fn main() {
    assert_eq!(merge_alternately("abc".into(), "pqr".into()), "apbqcr");
    assert_eq!(merge_alternately("abcd".into(), "pq".into()), "apbqcd");
    assert_eq!(merge_alternately("ab".into(), "pqrs".into()), "apbqrs");
}

File: 1768.py
def mergeAlternately(word1: str, word2: str) -> str:
    merged = []
    counter = 0

    while counter < len(word1) and counter < len(word2):
        merged.append(word1[counter])
        merged.append(word2[counter])
        counter += 1

    merged.append(word1[counter:])
    merged.append(word2[counter:])

    return "".join(merged)


print(mergeAlternately("abc", "pqr"))
print(mergeAlternately("ab", "pqrs"))
print(mergeAlternately("abcd", "pq"))
